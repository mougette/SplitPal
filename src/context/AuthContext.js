import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
