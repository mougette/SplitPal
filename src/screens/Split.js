import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const Split = ({navigation}) => {
  return (
    <View style={styles.master}>
      <Text style={styles.header}>TODO: Split Screen</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  master: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    fontSize: 32,
  },
});

export default Split;
