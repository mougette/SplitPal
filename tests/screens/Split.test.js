import React from 'react';
import renderer from 'react-test-renderer';
import { View, TextInput, Button, StyleSheet, TouchableOpacity, Text, ScrollView } from 'react-native';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import Split from '../../src/screens/Split'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve([
                                  {
                                    "TotalBalance": -85
                                  },
                                  {
                                    "FirstName": "2",
                                    "LastName": "2",
                                    "Email": "abc@e.com",
                                    "Balance": 0
                                  },
                                  {
                                    "FirstName": "fake",
                                    "LastName": "1",
                                    "Email": "fake1@gmail.com",
                                    "Balance": -19.5
                                  },
                                  {
                                    "FirstName": "fake",
                                    "LastName": "5",
                                    "Email": "fake5@gmail.com",
                                    "Balance": -23.5
                                  },
                                  {
                                    "FirstName": "hi",
                                    "LastName": "there",
                                    "Email": "newEmail@test.com",
                                    "Balance": 0
                                  },
                                  {
                                    "FirstName": "Sam",
                                    "LastName": "Hogenson",
                                    "Email": "shogenson@wisc.edu",
                                    "Balance": -40
                                  },
                                  {
                                    "FirstName": "test",
                                    "LastName": "today",
                                    "Email": "testing@test.com",
                                    "Balance": -2
                                  }
                                ]),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Split Tests', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Add and Remove buttons', async () => {
        const {getByText} = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="Split" component={Split} initialParams ={[
                {item: {
                FirstName: 'test',
                LastName: 'today',
                Email: 'testing@test.com',
                Balance: 'Split'
                }},
                {state:{email : "e@something.com"}},
                "NULL"]}/>
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
        await waitFor(() => {
        fireEvent.press(getByText("+"));
        fireEvent.press(getByText("Remove"));
        expect(1).toBe(1);
        })
        })
    it('Add and Remove buttons', async () => {
            const {getByText} = render(
            <AuthProvider>
            <NavigationContainer>
            <Stack.Navigator>
            <Stack.Screen name="Split" component={Split} initialParams ={[
                    {item: {
                    FirstName: 'test',
                    LastName: 'today',
                    Email: 'testing@test.com',
                    Balance: 'Split'
                    }},
                    {state:{email : "e@something.com"}},
                    "NULL"]}/>
            </Stack.Navigator>
            </NavigationContainer>
            </AuthProvider>);
            await waitFor(() => {
            fireEvent.press(getByText("Use Camera"));
            expect(1).toBe(1);
            })
            })

    it('Add entry and submit', async () => {
        const screen = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="Split" component={Split} initialParams ={[
                {item: {
                  FirstName: 'test',
                  LastName: 'today',
                  Email: 'testing@test.com',
                  Balance: 'Split'
                }},
                {state:{email : "e@something.com"}},
                "NULL"]}/>
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
        await waitFor(() => {
        fireEvent.press(screen.getByText("+"));
        fireEvent.changeText(screen.getAllByTestId("ItemName")[0], 'dress')
        fireEvent.changeText(screen.getAllByTestId("ItemName")[0], 'dresses')
        fireEvent.changeText(screen.getAllByTestId("ItemPrice")[0], '23')
        fireEvent.press(screen.getByText("Submit"));
        expect(1).toBe(1);
        })
        })
    it('ReRender', async () => {
        const screen = render(
        <Split {...{route:{params:[
                {item: {
                  FirstName: 'test',
                  LastName: 'today',
                  Email: 'testing@test.com',
                  Balance: 'Split'
                }},
                {state:{email : "e@something.com"}},
                "NULL"]}}}/>);

          screen.update(
            <Split {...{route:{params:[
                        {item: {
                          FirstName: 'test',
                          LastName: 'today',
                          Email: 'testing@test.com',
                          Balance: 'Split'
                        }},
                        {state:{email : "e@something.com"}},
                        {"items": [
                            [
                              "Sleeveless shirt",
                              "$19.99"
                            ],
                            [
                              "Faded jeans",
                              "$39.99"
                            ],
                            [
                              "Long dress",
                              "$67.99"
                            ],
                            [
                              "TAX 6%",
                              "$7.68"
                            ]
                          ]}]}}}/>
                );

        await waitFor(() => {
        expect(1).toBe(1);
        })
        })
});