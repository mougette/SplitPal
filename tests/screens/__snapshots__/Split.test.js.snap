// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Split Tests Add and Remove buttons 1`] = `undefined`;

exports[`Split Tests Add and Remove buttons 2`] = `undefined`;

exports[`Split Tests Add entry and submit 1`] = `undefined`;

exports[`Split Tests ReRender 1`] = `undefined`;
